src/mlpack/methods/ann/layer/sqnl_impl.hpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * @file methods/ann/layer/sqnl_impl.hpp
+ * @author Shaikh Yusuf Niaz
+ *
+ * Implementation of Square NonLinearity (SQNL) function as described by
+ * Wuraola, Adedamola and Patel, Nitish
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SQNL_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_SQNL_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "sqnl.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+SQNL<InputDataType, OutputDataType>::SQNL() 
+{
+    // Nothing to do here
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename OutputType>
+void SQNL<InputDataType, OutputDataType>::Forward(
+    const InputType& input, OutputType& output)
+{
+  output = arma::ones<OutputDataType>(arma::size(input));
+  for (size_t i = 0; i < input.n_elem; ++i)
+  {
+    if( input(i) > 2.0) 
```

Comment:
extra space after starting bracket.
and leave a space after the `if` statement, like so: 
```c++
if (input(i) > 2.0)
```

Reply:
Looks like we can skip this case, since `output` is already set to one, in which case we have to adjust the last `else`.